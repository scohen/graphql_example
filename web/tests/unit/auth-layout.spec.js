import { shallowMount } from '@vue/test-utils'
import AuthLayout from '@/components/auth-layout'

describe('auth-layout.vue', () => {
  it('renders the header slot', () => {
    const slots = {
      header: '<span>Hello There</span>'
    }
    const wrapper = shallowMount(AuthLayout, { slots })
    expect(wrapper.find('h1').text()).toBe('Hello There')
  })

  it('renders the default slot', () => {
    const slots = {
      default: '<div>Content Goes Here</div>'
    }
    const wrapper = shallowMount(AuthLayout, { slots })
    expect(wrapper.find('div.box').text()).toBe('Content Goes Here')
  })

  it('renders correctly', () => {
    const slots = {
      header: '<span>Hello There</span>',
      default: '<div>Content Goes Here</div>'
    }
    const wrapper = shallowMount(AuthLayout, { slots })
    expect(wrapper.html()).toMatchSnapshot()
  })
})
