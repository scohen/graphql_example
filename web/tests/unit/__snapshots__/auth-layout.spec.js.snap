// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`auth-layout.vue renders correctly 1`] = `
<section class="hero is-light is-fullheight">
  <div class="hero-body">
    <div class="container">
      <div class="columns is-centered">
        <div class="column is-6-tablet is-5-desktop is-4-widescreen">
          <h1 class="title has-text-centered"><span>Hello There</span></h1>
          <div class="box">
            <div>Content Goes Here</div>
          </div>
        </div>
      </div>
    </div>
  </div>
</section>
`;
